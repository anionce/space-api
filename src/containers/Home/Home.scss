@import '../../assets/styles/variables';

.home-container {
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
	margin: 36px;

	h2 {
		color: #fe6584;
	}

	img {
		height: 200px;
	}
}
