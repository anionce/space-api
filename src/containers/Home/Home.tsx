import React from 'react';
import { useTranslation } from 'react-i18next';
import './Home.scss';

export const Home = () => {
	const { t } = useTranslation();

	return (
		<div className='home-container'>
			<h1>{t('space_api.home.welcome')}</h1>
			<h2>{t('space_api.home.ready')}</h2>
		</div>
	);
};
