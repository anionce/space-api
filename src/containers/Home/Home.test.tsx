import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { screen } from '@testing-library/react';
import { customRender } from '../../assets/tests/utils';
import { Home } from './Home';

test('renders correcly', () => {
	const { container } = customRender(<Home />);

	expect(container).toMatchSnapshot();
});

test('all elements are displayed', () => {
	customRender(<Home />);

	expect(screen.getByText(/Welcome to the space API/i)).toBeInTheDocument();
	expect(screen.getByText(/Ready?/i)).toBeInTheDocument();
});
