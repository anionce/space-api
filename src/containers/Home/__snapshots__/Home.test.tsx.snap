// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correcly 1`] = `
<div>
  <div
    class="home-container"
  >
    <h1>
      Welcome to the space API
    </h1>
    <h2>
      Ready?
    </h2>
  </div>
</div>
`;
